### PR TITLE
Avoid incorrect OSGi dep on sslconfig.akka

### LINKED ISSUE #25579: Invalid OSGi bundle imports depending on com.typesafe.sslconfig.akka [0.2.3,1.0.0)
The com.typesafe.akka.stream bundle ends up importing com.typesafe.sslconfig.akka with a version range of [0.2.3,1.0.0), because of the range associated with com.typesafe.sslconfig.* in the OSGi setup.

A patch is forthcoming to fix this.

### PR DESCRIPTION
**Pull request by skitt**
Importing com.typesafe.sslconfig.* with the sslconfig version results
in OSGi bundle imports which also include com.typesafe.sslconfig.akka,
which can’t be satisfied with the given version range. Splitting up
the imported packages (com.typesafe.sslconfig,
com.typesafe.sslconfig.ssl.*, com.typesafe.sslconfig.util.*) results
in tighter-scoped imports which can be satisfied.

This fixes #25579.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @skitt,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Review by ktoso — COMMENTED** [review]
This seems okey. Thanks for the contrib

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
OK TO TEST

**Comment by raboof** [comment]
@skitt Thanks for the contribution! Could you look into signing the CLA? Should be a quick online process :)

**Comment by skitt** [comment]
Thanks for the reviews, @ktoso and @raboof. Regarding the CLA, I’m waiting for approval from our legal department.

**Comment by raboof** [comment]
Makes sense, staying tuned! Thanks!

**Comment by patriknw** [comment]
Have you verified that the contents of the jar files are the same aside from the expected changes when doing sbt publishLocal. The reason I'm cautious is that there have been too many times the osgi build plugin (or whatever) breaks the artifacts.

**Comment by skitt** [comment]
@patriknw I have; I’ve run two full builds, from scratch, with and without the patch, and used diffoscope to compare the results. The JARs only show differences in timestamps and generated docs; the classes are identical, as are the manifests, except where expected as a result of the patch.

**Review by patriknw — APPROVED** [review]
LGTM, thanks for verifying, then we are only waiting for CLA

**Comment by raboof** [comment]
PLS BUILD

**Comment by skitt** [comment]
> Thanks for the reviews, @ktoso and @raboof. Regarding the CLA, I’m waiting for approval from our legal department.

I’ve been given permission to sign the individual CLA and submit this under the terms of that CLA, including the terms of section 4. (So I’ve signed the individual CLA.)

**Comment by raboof** [comment]
Excellent, thanks for being responsible with this! I just checked and the CLA came through correctly (even if the github integration hasn't picked it up yet)

**Comment by patriknw** [comment]
This is one of latest OSGi contributions to Akka so I hope some in the Akka OSGi community is still listening here. As you know, in the core Akka team we don't have any OSGi expertise so we rely on the community to support and keep it up to date in Akka. Thanks so much for you help with this.

A related project is [sbt/sbt-osg](https://github.com/sbt/sbt-osgi), which is used by the Akka build. We would appreciate if you can help out with reviewing changes there so that contributions are not breaking things because of our lack of knowledge. Perhaps you could subscribe to notifications in that repository (should be low traffic)?

There is currently one pull request there that we would like such help with: https://github.com/sbt/sbt-osgi/pull/64

**Pull request merged**
